File: Modules/ServerTypes/EmployeeProfile/EmployeeCareerPathForm.ts
import { IntegerEditor, LookupEditor, DecimalEditor, DateEditor, TextAreaEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface EmployeeCareerPathForm {
    Id: IntegerEditor;
    CareerPathId: LookupEditor;
    EmployeeRowId: LookupEditor;
    NewValue: DecimalEditor;
    newDivision: LookupEditor;
    newDepartment: LookupEditor;
    newSection: LookupEditor;
    newOccupation: LookupEditor;
    newJobGrade: LookupEditor;
    newCostCentre: LookupEditor;
    CategoryId: IntegerEditor;
    careerPaathType: IntegerEditor;
    EffectiveDate: DateEditor;
    Description: TextAreaEditor;
    ManDesc: TextAreaEditor;
    oldValue: DecimalEditor;
    oldDivision: IntegerEditor;
    oldDepartment: IntegerEditor;
    oldSection: IntegerEditor;
    oldOccupation: IntegerEditor;
    oldJobGrade: IntegerEditor;
    oldCostCentre: IntegerEditor;
}

export class EmployeeCareerPathForm extends PrefixedContext {
    static formKey = 'EmployeeProfile.EmployeeCareerPath';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeCareerPathForm.init)  {
            EmployeeCareerPathForm.init = true;

            var w0 = IntegerEditor;
            var w1 = LookupEditor;
            var w2 = DecimalEditor;
            var w3 = DateEditor;
            var w4 = TextAreaEditor;

            initFormType(EmployeeCareerPathForm, [
                'Id', w0,
                'CareerPathId', w1,
                'EmployeeRowId', w1,
                'NewValue', w2,
                'newDivision', w1,
                'newDepartment', w1,
                'newSection', w1,
                'newOccupation', w1,
                'newJobGrade', w1,
                'newCostCentre', w1,
                'CategoryId', w0,
                'careerPaathType', w0,
                'EffectiveDate', w3,
                'Description', w4,
                'ManDesc', w4,
                'oldValue', w2,
                'oldDivision', w0,
                'oldDepartment', w0,
                'oldSection', w0,
                'oldOccupation', w0,
                'oldJobGrade', w0,
                'oldCostCentre', w0
            ]);
        }
    }
}

File: Modules/HumanResource/EmployeeProfile/EmployeeCareerPath/EmployeeCareerPathForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.EmployeeProfile.Forms;

[FormScript("EmployeeProfile.EmployeeCareerPath")]
[BasedOnRow(typeof(EmployeeCareerPathRow), CheckNames = true)]
public class EmployeeCareerPathForm
{
    [HideOnInsert,HideOnUpdate]
    public int Id { get; set; }


    [HalfWidth]
    public int CareerPathId { get; set; }
    [HalfWidth]
    public int EmployeeRowId { get; set; }

    public double NewValue { get; set; }

    public int newDivision { get; set; }
    public int newDepartment { get; set; }
    public int newSection { get; set; }
    public int newOccupation { get; set; }
    public int newJobGrade { get; set; }
    public int newCostCentre { get; set; }
    [HideOnInsert, HideOnUpdate]
    public int CategoryId { get; set; }
    [HideOnInsert, HideOnUpdate]
    public int careerPaathType    { get; set; }

    public DateTime EffectiveDate { get; set; }
   
   

    [TextAreaEditor(Rows = 4)]
    public string Description { get; set; }

    [TextAreaEditor(Rows = 4)]
    public string ManDesc { get; set; }





    public double oldValue { get; set; }

    public int oldDivision { get; set; }
    public int oldDepartment { get; set; }
    public int oldSection { get; set; }
    public int oldOccupation { get; set; }
    public int oldJobGrade { get; set; }
    public int oldCostCentre { get; set; }

}